### PR TITLE
feat(dashboard): add plain chat buttons

### PR DESCRIPTION
**Pull request by jainpawan21**
### What changed? Why was the change needed?
1. add plain chat buttons and other [customizations](https://www.plain.com/docs/chat/customization) in new dashbaord
2. fix [plain.init issue](https://novu-r9.sentry.io/issues/6108265818/?referrer=slack&notification_uuid=34cb66e4-aa36-499e-bd76-59860e370a0c&alert_rule_id=15115141&alert_type=issue)

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/src/components/header-navigation/customer-support-button.tsx
Comment (scopsy) [inline apps/dashboard/src/components/header-navigation/customer-support-button.tsx:1]: You can just ignore the whole file path on cspell

File: apps/dashboard/src/components/header-navigation/customer-support-button.tsx [hunk 1]
```diff
@@ -1,14 +1,140 @@
...
+import { RiQuestionFill } from 'react-icons/ri';
 import { HeaderButton } from './header-button';
+import { PLAIN_SUPPORT_CHAT_APP_ID } from '@/config';
+import { useAuth } from '@/context/auth/hooks';
 
 export const CustomerSupportButton = () => {
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  const { currentOrganization, currentUser } = useAuth();
+
+  const isLiveChatVisible =
```
Comment (scopsy) [inline apps/dashboard/src/components/header-navigation/customer-support-button.tsx]: As discussed, let's show this to all tiers.

File: apps/dashboard/src/components/header-navigation/customer-support-button.tsx [hunk 2]
```diff
@@ -1,14 +1,140 @@
...
+    currentOrganization?.apiServiceLevel !== 'free' &&
+    currentUser?.servicesHashes?.plain &&
```
Comment (scopsy) [inline apps/dashboard/src/components/header-navigation/customer-support-button.tsx]: Did we run a migraiton in the end for this?

File: apps/dashboard/src/components/header-navigation/customer-support-button.tsx [hunk 3]
```diff
@@ -1,14 +1,140 @@
...
+    setIsFirstRender(false);
+  }, [isLiveChatVisible, currentUser, isFirstRender]);
+
+  const showPlainLiveChat = () => {
+    if (isLiveChatVisible) {
+      try {
+        // @ts-ignore
+        window?.Plain?.open();
+      } catch (error) {
+        console.error('error opening plain chat', error);
```
Comment (scopsy) [inline apps/dashboard/src/components/header-navigation/customer-support-button.tsx]: Let's report this to sentry aswell Sentry.captureException

File: apps/web/src/components/layout/components/v2/HeaderNav.tsx [hunk 4]
```diff
@@ -46,34 +46,45 @@ export function HeaderNav() {
...
-          emailHash: currentUser?.servicesHashes?.plain,
-        },
-      });
+          customerDetails: {
+            email: currentUser?.email,
+            emailHash: currentUser?.servicesHashes?.plain,
+          },
+        });
+      } catch (error) {
+        console.error('error initializing plain chat', error);
```
Comment (scopsy) [inline apps/web/src/components/layout/components/v2/HeaderNav.tsx]: Sentry reporting

File: apps/web/src/components/layout/components/v2/HeaderNav.tsx [hunk 5]
```diff
@@ -46,34 +46,45 @@ export function HeaderNav() {
...
   const showLiveChat = () => {
-    if (currentUser?.servicesHashes?.plain && process.env.REACT_APP_PLAIN_SUPPORT_CHAT_APP_ID) {
-      // @ts-ignore
-      window.Plain.open();
+    if (isLiveChatVisible) {
+      try {
+        // @ts-ignore
+        window?.Plain?.open();
+      } catch (error) {
+        console.error('error opening plain chat', error);
```
Comment (scopsy) [inline apps/web/src/components/layout/components/v2/HeaderNav.tsx]: Sentry here too

File: apps/dashboard/src/components/header-navigation/customer-support-button.tsx (hunk 2 above)
Comment (jainpawan21) [reply apps/dashboard/src/components/header-navigation/customer-support-button.tsx]: yes both for US and EU

File: apps/dashboard/src/components/header-navigation/customer-support-button.tsx [hunk 6]
```diff
@@ -1,14 +1,145 @@
...
+              text: 'Contact Sales',
+              url: 'https://notify.novu.co/meetings/novuhq/novu-discovery-session-rr?utm_campaign=in_app_live_chat',
+            },
+          ],
+          entryPoint: 'default',
+          theme: 'light',
+          customerDetails: {
+            email: currentUser?.email,
+            emailHash: currentUser?.servicesHashes?.plain,
+          },
```
Comment (scopsy) [inline apps/dashboard/src/components/header-navigation/customer-support-button.tsx:47]: I think we can also pass here the name and external id for upsert

**Review by scopsy — APPROVED** [review]

**Pull request merged**